zephyr/net/mdns_responder.h does not support C++

### Describe the bug

When attempting to use mdns_responder_set_ext_records() from C++, linking fails because zephyr/net/mdns_responder.h does not have `extern "C" {..}`

### Regression

- [ ] This is a regression.

### Steps to reproduce

Add the following to samples/cpp/hello_world/prj.conf

- CONFIG_NETWORKING=y
- CONFIG_NET_HOSTNAME_ENABLE=y
- CONFIG_MDNS_RESPONDER=y
- CONFIG_DNS_SD=y
- CONFIG_MDNS_RESPONDER_DNS_SD=y

Add the following to samples/cpp/hello_world/src/main.cpp

- #include <zephyr/net/mdns_responder.h>
- Call mdns_responder_set_ext_records() from main()

### Relevant log output

```shell
zephyr-sdk-0.17.4/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.2.0/../../../../riscv64-zephyr-elf/bin/ld.bfd: app/libapp.a(main.cpp.obj): in function `.L0 ':
src/main.cpp:22: undefined reference to `mdns_responder_set_ext_records(dns_sd_rec const*, unsigned int)'
collect2: error: ld returned 1 exit status
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: macOS Tahoe
- SDK 0.17.4
- Zephyr v4.3.0

### Additional Context

This can be resolved with: 
```
diff --git a/include/zephyr/net/mdns_responder.h b/include/zephyr/net/mdns_responder.h
index 6ec9bb4c706..7a0f6a6f469 100644
--- a/include/zephyr/net/mdns_responder.h
+++ b/include/zephyr/net/mdns_responder.h
@@ -17,6 +17,10 @@
 #include <stddef.h>
 #include <zephyr/net/dns_sd.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /**
  * @brief Register continuous memory of @ref dns_sd_rec records.
  *
@@ -31,4 +35,8 @@
  */
 int mdns_responder_set_ext_records(const struct dns_sd_rec *records, size_t count);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif /* ZEPHYR_INCLUDE_NET_MDNS_RESPONDER_H_ */
```

--- a/include/zephyr/net/mdns_responder.h
+++ b/include/zephyr/net/mdns_responder.h
@@ -17,6 +17,10 @@
 #include <stddef.h>
 #include <zephyr/net/dns_sd.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /**
  * @brief Register continuous memory of @ref dns_sd_rec records.
  *
@@ -30,5 +34,9 @@
  * @return 0 for OK; -EINVAL for invalid parameters.
  */
 int mdns_responder_set_ext_records(const struct dns_sd_rec *records, size_t count);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* ZEPHYR_INCLUDE_NET_MDNS_RESPONDER_H_ */

